src/ensembl/src/content/app/browser/browser-messaging-service.ts

```diff
@@ -18,17 +18,103 @@ import windowService, {
   WindowServiceInterface
 } from 'src/services/window-service';
 
-import JSONValue from 'src/shared/types/JSON';
+import {
+  ZmenuLeavePayload,
+  ZmenuOutsideActivityPayload,
+  ZmenuEnterPayload
+} from 'src/content/app/browser/zmenu/zmenu-types';
+import {
+  ActivateBrowserPayload,
+  BrowserSetFocusLoationPayload,
+  BrowserSetFocusPayload
+} from 'src/content/app/browser/browserActions';
 
 /*
   This is a service for communicating between genome browser and React wrapper.
 */
 
+export enum BrowserToChromeMessagingActions {
+  ZMENU_CREATE = 'create_zmenu',
+  ZMENU_DESTROY = 'destroy_zmenu',
+  ZMENU_REPOSITION = 'update_zmenu_position'
+}
+
+export enum ChromeToBrowserMessagingActions {
+  ACTIVATE_BROWSER = 'activate_browser',
+  TOGGLE_TRACKS = 'toggle_tracks',
+  SET_FOCUS_LOCATION = 'set_focus_location',
+  SET_FOCUS = 'set_focus',
+  MOVE_LEFT = 'move_left',
+  MOVE_RIGHT = 'move_right',
+  MOVE_DOWN = 'move_down',
+  MOVE_UP = 'move_up',
+  ZOOM_BY = 'zoom_by',
+  PING = 'ping',
+  ZMENU_ENTER = 'zmenu-enter',
+  ZMENU_LEAVE = 'zmenu-leave',
+  ZMENU_ACTIVITY_OUTSIDE = 'zmenu-activity-outside' // TODO: sometime later, unify underscores vs hyphens (together with Genome Browser)
+}
```

Comment:
Let's not do all that in the browser-messaging-service file? Let's separate this into a separate `browser-message-creator` (or something) file that will be concerned only with the shape of the payload sent to the genome browser. Meanwhile the `browser-message-service` will only be concerned with actually sending and receiving messages.

Reply:
Yes, that's the actual plan. I'm just temporarily stacking it up in `browser-messaging-service`.